### PR TITLE
fix bugs in disk-expand

### PR DESCRIPTION
**Pull request by hopkiw**
* Include dmidecode (fixes confusing error message)
* Use correct function name for logging
* Don't bail on e2fsck errors; resize2fs includes an integrity pre check

### CODE REVIEW

File: packages/gce-disk-expand/packaging/gce-disk-expand.spec [hunk 1]
```diff
@@ -23,6 +23,8 @@ Source0: %{name}_%{version}.orig.tar.gz
 Requires: e2fsprogs, dracut, grep, util-linux, parted
```
Comment (zmarano) [inline packages/gce-disk-expand/packaging/gce-disk-expand.spec]: Requires: e2fsprogs, dracut, grep, util-linux, parted, gdisk

File: packages/gce-disk-expand/packaging/gce-disk-expand.spec [hunk 2]
```diff
@@ -23,6 +23,8 @@ Source0: %{name}_%{version}.orig.tar.gz
...
 Conflicts: dracut-modules-growroot
 
+BuildRequires: rsync
```
Comment (zmarano) [inline packages/gce-disk-expand/packaging/gce-disk-expand.spec]: We need rsync?

File: packages/gce-disk-expand/src/expandfs-lib.sh [hunk 3]
```diff
@@ -60,32 +60,31 @@ blkid_get_fstype() (
...
-    parted -m ---pretend-input-tty "$disk" print Fix >/dev/null 2>&1 </dev/null
-    if parted -sm "$disk" print 2>&1 | grep -q "fix the GPT"; then
-      echo "Failed to fix the GPT."
-      return 1
-    fi
-  fi
+  local label=$(sgdisk_get_label "$disk")
+  [ "$label" != "gpt" ] && return
+
+  sgdisk --move-second-header "$disk"
```
Comment (zmarano) [inline packages/gce-disk-expand/src/expandfs-lib.sh]: I believe you need to re-probe the partitions after this. At least that was what I found (partprobe).

File: packages/gce-disk-expand/src/usr/share/initramfs-tools/hooks/expand-rootfs [hunk 4]
```diff
@@ -36,3 +36,4 @@ copy_exec /sbin/e2fsck /bin
 copy_exec /sbin/blkid /bin
 copy_exec /sbin/resize2fs /bin
 copy_exec /bin/udevadm /bin
+copy_exec /usr/sbin/dmidecode /bin
```
Comment (zmarano) [inline packages/gce-disk-expand/src/usr/share/initramfs-tools/hooks/expand-rootfs]: What's dmidecode used for?

File: packages/gce-disk-expand/packaging/gce-disk-expand.spec (hunk 2 above)
Comment (hopkiw) [reply packages/gce-disk-expand/packaging/gce-disk-expand.spec]: yes, it's used in this file

File: packages/gce-disk-expand/src/expandfs-lib.sh (hunk 3 above)
Comment (hopkiw) [reply packages/gce-disk-expand/src/expandfs-lib.sh]: what did you find? we aren't changing partitions here so i didn't find this to be the case

File: packages/gce-disk-expand/src/usr/share/initramfs-tools/hooks/expand-rootfs (hunk 4 above)
Comment (hopkiw) [reply packages/gce-disk-expand/src/usr/share/initramfs-tools/hooks/expand-rootfs]: parted i think calls it and complains without it

File: packages/gce-disk-expand/src/expandfs-lib.sh (hunk 3 above)
Comment (zmarano) [reply packages/gce-disk-expand/src/expandfs-lib.sh]: It was several weeks back but I believe that you would still get the error that the header was wrong unless the kernel was forced to re-read the partition table. I believe this was when I was trying to do a check with sgdisk however.

File: packages/gce-disk-expand/src/usr/share/initramfs-tools/hooks/expand-rootfs (hunk 4 above)
Comment (zmarano) [reply packages/gce-disk-expand/src/usr/share/initramfs-tools/hooks/expand-rootfs]: Oh yes I think you are right.

**Comment by hopkiw** [comment]
working on:
debian-10 (gpt)
centos-7,rhel-7 (gpt, msdos)
centos-8,rhel-8 (gpt, msdos)

created 10g instance, installed package, resized disk to 30g, rebooted, confirmed 30g
this needs a test specifically to check that the condition where root is not the last partition does not get resized.

**Comment by hopkiw** [comment]
blocking on GoogleCloudPlatform/oss-test-infra#330

**Pull request merged**